src/textual/_work_decorator.py

```diff
@@ -7,23 +7,31 @@
 
 from functools import partial, wraps
 from inspect import iscoroutinefunction
-from typing import TYPE_CHECKING, Callable, Coroutine, TypeVar, Union, cast, overload
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Coroutine,
+    TypeVar,
+    Union,
+    cast,
+    overload,
+)
 
 from typing_extensions import ParamSpec, TypeAlias
 
 if TYPE_CHECKING:
     from .worker import Worker
 
 
-FactoryParamSpec = ParamSpec("FactoryParamSpec")
 DecoratorParamSpec = ParamSpec("DecoratorParamSpec")
 ReturnType = TypeVar("ReturnType")
 
 Decorator: TypeAlias = Callable[
     [
         Union[
             Callable[DecoratorParamSpec, ReturnType],
-            Callable[DecoratorParamSpec, Coroutine[None, None, ReturnType]],
+            Callable[DecoratorParamSpec, Coroutine[Any, Any, ReturnType]],
```

Comment:
Coroutines like `async def foo() -> T: ...` have their type inferred as `Coroutine[Any, Any, T]`, so I swapped all of the `Coroutine[None, None, T]` for `Coroutine[Any, Any, T]` under the assumption that the usage of `None` was a mistake.